src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1557,6 +1563,17 @@ protected virtual void ApplyCompilationOptionsChanged(ProjectId projectId, Compi
             this.OnCompilationOptionsChanged(projectId, options);
         }
 
+        /// <summary>
+        /// This method is called during <see cref="TryApplyChanges(Solution)"/> to change the default namespace.
+        ///
+        /// Override this method to implement the capability of changing default namespace.
+        /// </summary>
+        protected virtual void ApplyDefaultNamespaceChanged(ProjectId projectId, string? defaultNamespace)
+        {
+            Debug.Assert(CanApplyChange(ApplyChangesKind.ChangeProjectInfo));
```

Comment:
this new `ChangeProjectInfo` probably needs to be handled by `CanApplyChange` overrides in appropriate workspace implementations.  for example: [http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs,428](http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs,428)

Reply:
👍